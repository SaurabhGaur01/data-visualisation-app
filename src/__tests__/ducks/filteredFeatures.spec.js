import {
    setFilteredFeatures,
    clearFilteredFeatures,
    filteredFeaturesReducer,
} from '../../ducks/filteredFeatures';

describe('filteredFeatures Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setFilteredFeatures should match the snapshot' , () => {
            expect(setFilteredFeatures(mockObject)).toMatchSnapshot();
        });
        it('clearFilteredFeatures should match the snapshot' , () => {
            expect(clearFilteredFeatures(mockObject)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(filteredFeaturesReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(filteredFeaturesReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(filteredFeaturesReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(filteredFeaturesReducer('', setFilteredFeatures(mockObject))).toEqual(mockObject);
        });
        it('should reset the value to []' , () => {
            expect(filteredFeaturesReducer('', clearFilteredFeatures())).toEqual([]);
        });
    });
});