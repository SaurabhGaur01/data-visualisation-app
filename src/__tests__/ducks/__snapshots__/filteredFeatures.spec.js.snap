// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filteredFeatures Duck Tests Action Generator clearFilteredFeatures should match the snapshot 1`] = `
Object {
  "type": "modules/data-visualisation-app/CLEAR_FILTERED_FEATURES",
}
`;

exports[`filteredFeatures Duck Tests Action Generator setFilteredFeatures should match the snapshot 1`] = `
Object {
  "data": "mockValue",
  "type": "modules/data-visualisation-app/SET_FILTERED_FEATURES",
}
`;
